pkg/commands/addmetadata_test.go

```diff
@@ -0,0 +1,226 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package commands
+
+import (
+	"testing"
+
+	"strings"
+
+	"github.com/kubernetes-sigs/kustomize/pkg/constants"
+	"github.com/kubernetes-sigs/kustomize/pkg/fs"
+)
+
+func TestLabelsValid(t *testing.T) {
```

Comment:
a cleaner way to set up the tests is

1) one test for the common Validate method.  Cover the conversion of the command line string into a valdi map[string]string
2) a test for the addLabels routine which takes a map[string]string as an arg
3) a test for addAnnotations, which also takes a map[s]s.

There's too much duplication in what's below, and it doesn't take advantage of the Validate/runHandler decoupling.

it's good coverage - just too much code.